### ISSUE TITLE
PSOGPU doesn't work with `mutable struct NonlinearProblem`

### ISSUE THREAD

**Issue by utkarsh530** [SciML/PSOGPU.jl#48]
`HybridPSO` won't work with mutable structs, similar to something we saw in `DiffEqGPU`. 
https://github.com/SciML/SciMLBase.jl/blob/599174668d1964cef610eec6c0652a2383f2e374/src/problems/nonlinear_problems.jl#L155

Should we do an internal conversion like `struct ImmutableNonlinearProblem` under the hood? @ChrisRackauckas

**Comment by ChrisRackauckas** [comment]
Yes just like with ODEProblem you'll need to specialize it out of the kernel.
